app/channel/utility/src/main/java/org/phoebus/channelfinder/utility/JobRunnableWithCancel.java

```diff
@@ -0,0 +1,87 @@
+package org.phoebus.channelfinder.utility;
+
+import org.phoebus.framework.jobs.JobMonitor;
+import org.phoebus.framework.jobs.JobRunnable;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+import java.util.function.BiConsumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A helper class which provides the framework for running tasks in the background while still monitoring for cancel requests.
+ *
+ * @author Kunal Shroff
+ */
+public abstract class JobRunnableWithCancel implements JobRunnable {
+
+    private static final Logger logger = Logger.getLogger(JobRunnableWithCancel.class.getName());
+    private static final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Consider using `org.phoebus.framework.jobs.NamedThreadFactory`:
 1) That way, the thread will have a recognizable name
 2) The thread will be a daemon thread, allowing the app to exit

Reply:
I have used the NamedThreadFactory, not sure I have the best name for the time being

Reply:
Sure, that's fine. When you see a thread named "JobRunnableWithCancel-1", you can locate that in the sources. With "pool-05-thread-03" it's a lot harder.